lib/identifiers/isbn.rb

```diff
@@ -39,11 +39,25 @@ class ISBN
       \d{1,7}   # ISBN title enumerator and check digit
       \b
     }x
+    TEXT_AFTER_PREFIX_REGEXP = ':?\s*(\d.*)$'.freeze
+
+    def self.extract(str , prefixes = [])
+      str = match_strings_with_prefix(str , prefixes) if prefixes.any?
 
-    def self.extract(str)
       extract_isbn_as(str) + extract_thirteen_digit_isbns(str) + extract_ten_digit_isbns(str)
     end
 
+    def self.match_strings_with_prefix(str, prefixes)
+      prefix_regexp = prefixes.join('|')
+
+      str
+        .to_s
```

Comment:
Do we need a `to_s` here since `str` should represent a string?

Reply:
We have that in all other methods so I kept it just in case, not to break anything. It will respond to any object that implements `.to_s`, not only strings and being Open Source we don't know how people are using it @orthodoX

Reply:
Right it makes sense 👍🏻